users/dierkes/preprocessing/segment_cutting.py

```diff
@@ -15,42 +18,92 @@ class CutAndStitchSpeechSegmentsFromCorpusJob(Job):
         :param int n_workers: number of threads for work on files in parallel
         :param str file_extension: currently supported are wav, flac and mp3
         """
         self.bliss_corpus_file = bliss_corpus_file
+        self.min_length = min_length
         self.target_length = target_length
         self.n_workers = n_workers
         self.file_extension = file_extension
-        assert file_extension in ['wav', 'flac', 'mp3']
+        assert file_extension in ["wav", "flac", "mp3"]
 
         self.out_audio_path = self.output_path("audio/", directory=True)
-        self.rqmt = {'time': 8, 'mem': 16, 'cpu': self.n_workers}
+        self.out_length_hist = self.output_path("file_length_hist.png")
+        self.cut_and_stitch_segments_rqmt = {
+            "time": 8,
+            "mem": 16,
+            "cpu": self.n_workers,
+        }
+        self.length_dist_rqmt = {"time": 2, "mem": 1, "cpu": self.n_workers}
 
     def tasks(self):
-        yield Task("run", rqmt=self.rqmt)
+        yield Task("cut_and_stitch_segments", rqmt=self.cut_and_stitch_segments_rqmt)
+        yield Task("plot_length_distribution", rqmt=self.length_dist_rqmt)
 
-    def run(self):
+    def cut_and_stitch_segments(self):
         self.corpus_object = corpus.Corpus()
         self.corpus_object.load(self.bliss_corpus_file.get_path())
         recordings = list(self.corpus_object.all_recordings())
 
         print(f"{len(recordings)} recordings detected")
         print(f"launching {self.n_workers} processes")
 
-        tasks = [(r, self.out_audio_path, self.target_length, self.file_extension) for r in recordings]
+        tasks = [(r, self.out_audio_path.get_path(), self.file_extension) for r in recordings]
         with multiprocessing.Pool(processes=self.n_workers) as pool:
             for i, _ in enumerate(pool.imap_unordered(self.cut_file, tasks)):
                 if i % 100 == 0:
                     logging.info(f"{i} of {len(tasks)} files done")
 
+    def plot_length_distribution(self):
+        files = [
+            f"{self.out_audio_path.get_path()}/{f}"
+            for f in os.listdir(self.out_audio_path.get_path())
+            if os.path.splitext(f)[1] == f".{self.file_extension}"
+        ]
+        num_files = len(files)
+
+        with multiprocessing.Pool(processes=self.n_workers) as pool:
+            file_lengths = list(pool.imap_unordered(self.get_length, files, 100))
+
+        avg_length = round(sum(file_lengths) / num_files, 2)
+        length = round(sum(file_lengths) / 3600, 2)
+        fig, ax = plt.subplots()
+        ax.hist(file_lengths, bins=80)
+        fig.text(
+            0.5,
+            0.9,
+            f"Average length: {avg_length}s, Summed length: {length}h",
+            ha="center",
+            va="center",
+        )
+        ax.set_ylabel("count")
+        ax.set_xlabel("file length")
+        plt.savefig(self.out_length_hist.get_path())
+
+    def get_length(self, file):
```

Comment:
I suppose this doesn't need to be a class method (it's static also by the way). You can just define it inside the function at the place where you need it.

Reply:
I wasn't aware of this, then I prefer the old version (but as static method). Sorry...